src/Orleans.EventSourcing.Snapshot/SnapshotStateWithMetaData.cs

```diff
@@ -46,9 +46,9 @@ public override string ToString()
 
     public interface ISnapshotMetaData
```

Comment:
@codrut51 I have a question, Under which situation you use this interface?

Reply:
@zh6335901 When using a custom grain storage provider this interface can be used to cast the incoming state and write some logic to ensure no empty snapshots are stored. With the current implementation, the snapshot is stored into the default grain provider even when it is empty/null. By using this interface a developer doesn't need to know the snapshot state or entry generics and can write custom logic in a custom grain storage provider to prevent unnecessary storage of data.